src/loaders/stm_load.c

```diff
@@ -270,7 +270,8 @@ static int stm_load(struct module_data *m, HIO_HANDLE * f, const int start)
 	D_(D_INFO "Stored samples: %d", mod->smp);
 
 	for (i = 0; i < mod->ins; i++) {
-		if (mod->xxs[i].len > 1) {
+		if (sfh.ins[i].volume && sfh.ins[i].length) {
```

Comment:
As you can see in [code](http://pastebin.com/c5kcFPHA), ST2 treats volume 0 as a flag of empty sample. Much more important, that the ST2 does not load samples as a continuous stream, but uses sfh.ins[i].rsvd1 to calculate offset (look at seg003:18D1 in code) from begining of the file. libxmp (and other libraries) does not do that and corrupt samples while loading, you can hear it on wonder.stm and other modules with non serial sample arrangement.